eth/executionclient/execution_client.go

```diff
@@ -262,9 +266,22 @@ func (ec *ExecutionClient) Healthy(ctx context.Context) error {
 	if sp != nil {
 		ec.logger.Error("Execution client is not synced")
 		ec.metrics.ExecutionClientSyncing()
-		if sp.CurrentBlock < sp.HighestBlock-ec.syncTolerance {
+
+		hblock := sp.HighestBlock > ec.syncDistanceTolerance // make sure we don't underflow
+		blockOutOfLimit := sp.CurrentBlock < sp.HighestBlock-ec.syncDistanceTolerance
+
+		// block out of sync distance tolerance
+		blockOutOfSDT := hblock && blockOutOfLimit
+
+		// maximum time we can tolerate since last Syncing=False
+		timeOutOfSDT := ec.syncLastSuccess.Before(time.Now().Add(-syncTimeTolerance))
+
+		// in both cases we crash
+		if blockOutOfSDT || timeOutOfSDT {
```

Comment:
This could be simplified for readability, and also adding information to errors:
```suggestion
	syncDistance := max(sp.HighestBlock, sp.CurrentBlock) - sp.CurrentBlock
	if syncDistance > ec.syncDistanceTolerance {
		return fmt.Errorf("sync distance exceeds tolerance (%d): %w", syncDistance, errSyncing)
	}
	
	if ec.syncLastSuccess.Before(time.Now().Add(-syncTimeTolerance)) {
		return fmt.Errorf("not synced for too long (%s): %w", time.Since(ec.syncLastSuccess), errSyncing)
	}
```

Reply:
@anatolie-ssv @MatusKysel readability is subjective, but the reason i like this is because it more naturally reflects how we think:

- "crash if sync distance > X slots"
- "crash if not synced for over X time"

Reply:
integrated these changes and also inverted the `syncLastSuccess => syncLastUnsuccessful` to simplify the happy path.